FreeAPS/Sources/Shortcuts/Bolus/BolusIntent.swift

```diff
@@ -0,0 +1,80 @@
+import AppIntents
+import Foundation
+import Intents
+import Swinject
+
+@available(iOS 16.0,*) struct BolusIntent: AppIntent {
+    // Title of the action in the Shortcuts app
+    static var title = LocalizedStringResource("Enact Bolus", table: "ShortcutsDetail")
+
+    // Description of the action in the Shortcuts app
+    static var description = IntentDescription(.init("Allow to send a bolus to the app", table: "ShortcutsDetail"))
+
+    internal var bolusRequest: BolusIntentRequest
+
+    init() {
+        bolusRequest = BolusIntentRequest()
+    }
+
+    @Parameter(
+        title: LocalizedStringResource("Amount", table: "ShortcutsDetail"),
+        description: LocalizedStringResource("Bolus amount in U", table: "ShortcutsDetail"),
+        controlStyle: .field,
+        inclusiveRange: (lowerBound: 0, upperBound: 200),
+        requestValueDialog: IntentDialog(LocalizedStringResource(
+            "What is the value of the bolus amount in insulin units ?",
+            table: "ShortcutsDetail"
+        ))
+    ) var bolusQuantity: Double?
+
+    @Parameter(
+        title: LocalizedStringResource("Confirm Before applying", table: "ShortcutsDetail"),
+        description: LocalizedStringResource("If toggled, you will need to confirm before applying", table: "ShortcutsDetail"),
+        default: true
+    ) var confirmBeforeApplying: Bool
+
+    static var parameterSummary: some ParameterSummary {
+        When(\.$confirmBeforeApplying, .equalTo, true, {
+            Summary("Applying \(\.$bolusQuantity) U ", table: "ShortcutsDetail") {
+                \.$confirmBeforeApplying
+            }
+        }, otherwise: {
+            Summary("Immediately applying \(\.$bolusQuantity) U", table: "ShortcutsDetail") {
+                \.$confirmBeforeApplying
+            }
+        })
+    }
+
+    @MainActor func perform() async throws -> some ProvidesDialog {
+        do {
+            guard let amount: Double = bolusQuantity else {
```

Comment:
In order to allow users to input the bolus amount (as noted by @mountrcg https://github.com/nightscout/Trio/pull/177#issuecomment-2116948179), we need to make `bolusQuantity` mandatory, and not optional. See the comment above at line 28.

Since `bolusQuantity` is no longer optional, we must remove this `guard` and `else {...}` here.

So, change this line from:
`guard let amount: Double = bolusQuantity else {`

To:
`let amount: Double = bolusQuantity`

We then need to remove the contents of the `else {...}`.

So, delete lines 51, 52, 53, 54, 55, 56, 57:
```
                return .result(
                    dialog: IntentDialog(LocalizedStringResource(
                        "Amount not correctly defined",
                        table: "ShortcutsDetail"
                    ))
                )
            }
```

See my comment https://github.com/nightscout/Trio/pull/177#issuecomment-2127028568 confirming the same.

Reply:
I committed this to the branch and tested on a physical device 👍